dashboard/src/components/Config/AppRepoList/AppRepoForm.test.tsx

```diff
@@ -123,5 +178,30 @@ describe("when the repository info is already populated", () => {
         authMethod: "bearer",
       });
     });
+
+    it("should pre-select the existing docker registry secret", () => {
+      const secret = {
+        metadata: {
+          name: "foo",
+        },
+      } as ISecret;
+      const repo = { metadata: { name: "foo" }, spec: { dockerRegistrySecrets: ["foo"] } } as any;
+      const wrapper = shallow(
+        <AppRepoForm {...defaultProps} imagePullSecrets={[secret]} repo={repo} />,
+      );
+      expect(wrapper.state()).toMatchObject({ selectedImagePullSecrets: { foo: true } });
+
+      // Add new secret, force re-render
+      const secret2 = {
+        metadata: {
+          name: "bar",
+        },
+      } as ISecret;
+      wrapper.setProps({ imagePullSecrets: [secret, secret2] });
+
+      expect(wrapper.state()).toMatchObject({
+        selectedImagePullSecrets: { foo: true, bar: false },
+      });
+    });
```

Comment:
We'll need to think about what to do if the existing secret referenced by the app repo has been deleted (I'm erroring during the deploy in this case).

Reply:
We are only listing in the form existing secrets. If an AppRepository references a secret that no longer exists, when updating it with the dashboard, it will be automatically removed.

Reply:
Right, that's what I think could be an issue: Scenario: User 1 adds an app repository with a specific secret. User two not using kubeapps at some point later kubectl delete's the secret . User 1 later edits the app repo and *silently* the secret is removed as a requirement of the app repo, without them noticing. People deploy charts and they fail.

I just wonder if it'd be better to make the user aware: "This app repository referred to the secret 'my-private-repo-creds' which is no longer present. Please ensure secrets are present for all private registries used" or something?

Reply:
I agree but that would be a weird corner case, it's not common for people to edit an apprepository that's working for a while. I'm sure that people would notice before because the images referenced by the charts won't be pulled anymore.